### PR TITLE
issue146 Allow JAX-RS endpoints Complete/Compensate to be invoked usi…

### LINKED ISSUE #146: Allow JAX-RS endpoints Complete/Compensate to be invoked using any HTTP method
When the Compensate or Complete methods are JAX-RS endpoints they should be allowed to be annotated with any HTTP method, ie the following spec requirement is too restrictive:

> The application developer indicates which JAX-RS method to use for a compensating
activity by annotating it with both the `@Compensate` and the JAX-RS `@PUT` annotations.

and

> ... If the annotation is not accompanied by
> a JAX-RS `@PUT` annotation the error SHOULD be reported using a JAX-RS
> exception mapper that maps to a `412 Precondition Failed` HTTP status code.

### PR DESCRIPTION
**Pull request by xstefank**
…ng any HTTP method

resolves #146 

Signed-off-by: xstefank <xstefank122@gmail.com>

### CODE REVIEW

**Comment by mmusgrov** [comment]
We should have at least one test that uses an HTTP method that is different from the recommended HTTP method:

- `@PUT` `@Forget`
- `@PUT` `@Status`
- `@Compensate` `@GET` 
- `@Complete` `@POST`

**Comment by xstefank** [comment]
+1, I was thinking whether we want to have it in TCK. I will update exisiting test.

**Comment by xstefank** [comment]
@mmusgrov I've update ContextTckResource with your suggestions

File: tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java [hunk 1]
```diff
@@ -345,7 +345,7 @@ public Response leave(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
         return Response.ok().entity(lraId).build();
     }
 
-    @PUT
+    @GET
```
Comment (mmusgrov) [inline tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/ContextTckResource.java]: This is is quite a big change to the spec so I would advise that we at least include a comment, something like:

> @GET // @PUT SHOULD be used but verify that other HTTP methods are supported on compensations (and ditto for the completion, status and forget callbacks).

**Comment by mmusgrov** [comment]
@rdebusscher Do you mind if I merge Martin's PR?

**Pull request merged**